### PR TITLE
viewer toolbar: allow clicking carrot to show submenu

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request adds support for clicking on the carrot to open the viewer toolbar's submenu, along with edits to the tooltip and a hover animation to help aid whether the button or the carrot is being clicked.

https://github.com/spacetelescope/jdaviz/assets/877591/67c8aba9-1de6-4679-81b8-355c259233fb

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/components/toolbar_nested.vue [hunk 1]
```diff
@@ -5,10 +5,10 @@
             <template v-slot:activator="{ on }">
                 <v-btn v-on="on" icon :value="id" style="min-width: 40px !important" @contextmenu="(e) => show_submenu(e, has_suboptions, menu_ind)">
                     <img class="invert-if-dark" :src="img" width="20px" @click.ctrl.stop=""/>
-                    <v-icon small v-if="has_suboptions" class="suboptions-carrot invert-if-dark" @click.ctrl.stop="">mdi-menu-down</v-icon>
+                    <v-icon small v-if="has_suboptions" class="suboptions-carrot invert-if-dark" @click="(e) => show_submenu(e, has_suboptions, menu_ind)" @click.ctrl.stop="">mdi-menu-down</v-icon>
                 </v-btn>
             </template>
-            <span>{{ tooltip }}{{has_suboptions ? " [right-click for alt. tools]" : ""}}</span>
+            <span>{{ tooltip }}{{has_suboptions ? " [click arrow for alt. tools]" : ""}}</span>
```
Comment (pllim) [inline jdaviz/components/toolbar_nested.vue:11]: This is a behavior change, no? Does this need a change log, for those who are used to right-clicking?

Comment (kecnry) [reply jdaviz/components/toolbar_nested.vue:11]: Right clicking is still supported, just not advertised in the tooltip anymore

**Review by pllim — APPROVED** [review]
Black magic.

**Review by rosteen — APPROVED** [review]
LGTM, I like it more than I initially thought I would. Feels intuitive 👍

**Pull request merged**